tls/s2n_x509_validator.c

```diff
@@ -701,6 +708,23 @@ static S2N_RESULT s2n_x509_validator_verify_cert_chain(struct s2n_x509_validator
      */
     X509_STORE_CTX_set_flags(validator->store_ctx, X509_V_FLAG_PARTIAL_CHAIN);
 
+#ifdef S2N_LIBCRYPTO_SUPPORTS_CUSTOM_OID
```

Comment:
I'd recommend moving this to a separate function to try and avoid making the verify_cert_chain function even longer. Especially since it requires the `#if/else`.